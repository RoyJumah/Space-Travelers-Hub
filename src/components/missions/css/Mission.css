* {
  box-sizing: border-box;
}

.success-message {
  color: greenyellow;
}

.mission-table {
  width: 80%;
  border-collapse: collapse;
  margin: 50px;
}

.mission-table th,
.mission-table td {
  border: 1px solid #ddd;
  padding: 8px;
  text-align: left;
}

.mission-table td {
  width: 25%;
}

.mission-table td:nth-child(2) {
  width: 50%;
}

.mission-table th {
  text-align: center;
  background-color: #fff !important;
}

.mission-table td:first-child {
  text-align: center;
  font-size: 24px;
  font-weight: bold;
  width: 15%;
}

tr:nth-child(odd) {
  background-color: #eee;
}

.mission-table td:nth-child(3) {
  width: 15%;
}

.btn {
  padding: 8px 10px;
  background-color: inherit;
}

.failure__button {
  color: red;
  border-color: red;
}

.not__member {
  text-transform: uppercase;
  background-color: #999999;
  color: white;
  font-size: 16px;
}

.active__member {
  text-transform: capitalize;
  color: #fff;
  background-color: #00b300;
}

.active__member,
.not__member {
  border-radius: 5px;
  padding: 4px;
  display: inline-block;
  font-weight: bold;
}
