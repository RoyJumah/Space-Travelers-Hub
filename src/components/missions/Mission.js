/* eslint-disable camelcase */
import PropTypes from 'prop-types';
import './css/Mission.css';
import { useDispatch } from 'react-redux';
import React, { useState } from 'react';
// import store from '../../redux/ConfigureStore';
import { joinMission, leaveMission } from '../../redux/missions/missionsSlice';

const Mission = props => {
  const { mission } = props;
  const { mission_id, mission_name, description } = mission;
  const dispatch = useDispatch();

  const [success, setSuccess] = useState(false);

  const handleJoinMission = () => {
    dispatch(joinMission({ id: mission_id }));
    // console.log(store.getState());
    setSuccess(true);
  };

  const handleLeaveMission = () => {
    dispatch(leaveMission({ id: mission_id }));
    // console.log(store.getState());
  };

  return (
    <div className="mission-list">
      <h1>{mission_name}</h1>
      <p>{description}</p>
      <button type="button" onClick={handleJoinMission}>
        Join Mission
      </button>
      {success && (
        <p className="success-message">Successfully joined the mission! ðŸš€</p>
      )}

      <button type="button" onClick={handleLeaveMission}>
        Leave Mission
      </button>
    </div>
  );
};

Mission.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    mission_name: PropTypes.string,
    description: PropTypes.string,
  }).isRequired,
};

export default Mission;
