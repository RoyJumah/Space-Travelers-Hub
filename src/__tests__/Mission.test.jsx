import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Mission from '../components/missions/Mission';
import { Provider } from 'react-redux';
import store from '../redux/ConfigureStore';

const mission = {mission_id: '1', mission_name: 'Thaicom', description: 'description', joined: false}

describe('Display mission component', () => {
  test('it matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Mission mission = {mission} />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('should return  Mission Component', () => {
    render(
      <Provider store={store}>
        <Mission mission = {mission} />
      </Provider>
    );
  })
})
