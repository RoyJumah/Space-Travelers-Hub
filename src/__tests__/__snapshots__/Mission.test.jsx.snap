// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display mission component it matches snapshot 1`] = `
<table>
  <tbody>
    <tr>
      <td>
        Thaicom
      </td>
      <td>
        description
      </td>
      <td>
        <p
          className="not__member"
        >
          Not a member
        </p>
      </td>
      <td>
        <button
          className=" btn success__button"
          onClick={[Function]}
          type="button"
        >
          Join Mission
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
